orchestrator/do-release/Jenkinsfile

```diff
@@ -64,18 +64,30 @@ def doRelease(exoUser, jenkinsAgentRootPath, jiraID, project, releaseCMD, isInPa
 // Read Datas from JSON Catalog and executes releases
 def doReleases(exoUser, jenkinsAgentRootPath, jiraID, projectsToRelease, releaseCMD, isInParallel) {
 
-   def JSONCatalog = new URL("${CATALOG_BASE_URL}/${JIRA_ID}.json")
-   def catalog = jsonParser(JSONCatalog.newReader())
+    def JSONCatalog = new URL("${CATALOG_BASE_URL}/${JIRA_ID}.json")
+    def catalog = jsonParser(JSONCatalog.newReader())
 
     echo "Number of Projects in Catalog: ${catalog.size}"
     // Loop first on projectsToRelease to keep the order
     for (i = 0; i < projectsToRelease.length; i++) {
-         def projectName = projectsToRelease[i];
-         for (j = 0; j < catalog.size; j++) {
-             if (catalog[j].name == projectName) {
+        def projectName = projectsToRelease[i];
+        def found = false
+        for (j = 0; j < catalog.size; j++) {
+            if (catalog[j].name == projectName) {
+                if (found) {
+                    echo "ERROR The project ${projectName} was found several times in the catalog file"
+                    currentBuild.result = "FAIL"
+                    return
+                }
+                found = true
```

Comment:
Shouldn't we do these checks before starting any release ? It can save time, for example in the case of a project defined twice in the catalog and the second one is the good one, the first project has been released for nothing.

Reply:
You are right, the faster the problems are detected the faster to solve they will be.
it's changed